task/project/task/serializers.py

```diff
@@ -1,10 +1,37 @@
 from rest_framework import serializers
 
 from .models import TaskModel
+from app.models import CustomUser, UserProfile
+from app.serializerls import UserProfileSerializer, UserSerializer
 
 
-class CreateTaskSerializers(serializers.ModelSerializer):
-
+class TaskSerializers(serializers.ModelSerializer):
     class Meta:
         model = TaskModel
         fields = "__all__"
+
+
+class TaskDetailsSerializer(serializers.ModelSerializer):
+    assignee = serializers.SerializerMethodField()
+    reporter = serializers.SerializerMethodField()
+
+    class Meta:
+        model = TaskModel
+        fields = ("task", "assignee", "reporter")
+
+    def get_assignee(self, obj):
+        obj = obj.assigned_by
+        data = UserProfile.objects.get(user=obj)
```

Comment:
we have user instance already no need to make extra query here.
data = UserProfile.objects.get(user=obj)